cogs/roles.py

```diff
@@ -17,129 +15,228 @@
 # You should have received a copy of the GNU General Public License
 # along with Canary. If not, see <https://www.gnu.org/licenses/>.
 
-# discord-py requirements
 import discord
+
 from discord import utils
 from discord.ext import commands
+from enum import Enum
+from typing import Optional, Tuple
+
+from .utils.checks import is_moderator
 from .utils.paginator import Pages
 
 
+class RoleTransaction(Enum):
+    ADD = "add"
+    REMOVE = "remove"
+
+
 class Roles(commands.Cog):
+    ALL_CATEGORIES = (
+        "pronouns",
+        "fields",
+        "faculties",
+        "years",
+        "generics",
+    )
+
+    CATEGORY_LIMITS = {
+        "pronouns": None,
+        "fields": 2,
```

Comment:
Limit here should be 3 IMO. Arts is major + 2 minors.

Reply:
5c27cd3eb5efe0b0e56c02415ac994957a49cfea